app/tests/archives_tests/test_views.py

```diff
@@ -999,3 +1001,183 @@ def test_archive_item_bulk_delete_permissions(client):
         list(response.context["form"].fields["civ_sets_to_delete"].queryset)
         == []
     )
+
+
+@pytest.mark.django_db
+@pytest.mark.parametrize(
+    "distractor_generator",
+    (
+        # Different CIVs in distractor
+        lambda civs: [ComponentInterfaceValueFactory()],
+        # Partial CIVs in distractor
+        lambda civs: [civs[0]],
+        # Extra CIVs in distractor
+        lambda civs: [*civs, ComponentInterfaceValueFactory()],
+    ),
+)
+def test_archive_item_job_list_filter(distractor_generator, client):
+    archive = ArchiveFactory()
+    archive.add_editor(editor := UserFactory())
+
+    ai = ArchiveItemFactory(archive=archive)
+    civs = [
+        ComponentInterfaceValueFactory(),
+        ComponentInterfaceValueFactory(),
+    ]
+    ai.values.set(civs)
+
+    aj1 = AlgorithmJobFactory(public=True)
+    aj2 = AlgorithmJobFactory(public=True)
+    for aj in aj1, aj2:
+        aj.inputs.set(ai.values.all())
+
+    distractor = AlgorithmJobFactory(public=True)
+    distractor_civs = distractor_generator(civs)
+    distractor.inputs.set(distractor_civs)
+
+    # Sanity
+    sort_column = 1
+    assert ArchiveItemJobListView.columns[sort_column].sort_field == "created"
+
+    response = get_view_for_user(
+        viewname="archives:item-job-list",
+        client=client,
+        method=client.get,
+        reverse_kwargs={
+            "slug": archive.slug,
+            "pk": ai.pk,
+        },
+        user=editor,
+        follow=True,
+        data={
+            "length": 10,
+            "draw": 1,
+            "order[0][dir]": "asc",
+            "order[0][column]": sort_column,
+        },
+        **{"HTTP_X_REQUESTED_WITH": "XMLHttpRequest"},
+    )
+
+    assert response.status_code == 200  # Sanity
+
+    resp = response.json()
+
+    data = resp["data"]
+
+    assert len(data) == 2, "distractor is not selected"
+
+    assert resp["recordsTotal"] == 2, "page length has all items"
+
+    # Sanity
+    detail_column = 0
+    assert ArchiveItemJobListView.columns[detail_column].title == "Detail"
+
+    assert str(aj1.pk) in resp["data"][0][detail_column]
+    assert str(aj2.pk) in resp["data"][1][detail_column]
+
+
+@pytest.mark.django_db
+def test_archive_item_job_list_permissions_filter(client):
+    archive = ArchiveFactory()
+    archive.add_editor(editor := UserFactory())
```

Comment:
That was incredibly confusing. There is no point at all to using a walrus argument here.

```suggestion
    editor = UserFactory()
    archive.add_editor(editor)
```